NuCivic/dkan
# Fix broken update that tries to update fields that may not exist.

## Description

A recent update hook implementation in dkan_dataset attempts to modify fields without first checking that they exist. This PR just adds checks to ensure the fields exist.

## How to reproduce

1. Install a DKAN site on a codebase prior to https://github.com/NuCivic/dkan/pull/1387, _without_ enabling the "odfe" module(s).
2. Update to the latest release.
3. Observe the following error:
```
    Executing dkan_dataset_update_7002
    Undefined index: field_name field.crud.inc:222
    Attempt to update a non-existent field.
```
